src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -181,44 +184,12 @@ void GcsResourceManager::HandleReportResourceUsage(
   ++counts_[CountType::REPORT_RESOURCE_USAGE_REQUEST];
 }
 
-// TODO(rickyx): We could update the cluster resource manager when we update the load
-// so that we will no longer need node_resource_usages_.
-std::unordered_map<google::protobuf::Map<std::string, double>, rpc::ResourceDemand>
-GcsResourceManager::GetAggregatedResourceLoad() const {
-  std::unordered_map<google::protobuf::Map<std::string, double>, rpc::ResourceDemand>
-      aggregate_load;
-  if (node_resource_usages_.empty()) {
-    return aggregate_load;
-  }
-  for (const auto &usage : node_resource_usages_) {
-    // Aggregate the load reported by each raylet.
-    FillAggregateLoad(usage.second, &aggregate_load);
-  }
-  return aggregate_load;
-}
-
-void GcsResourceManager::FillAggregateLoad(
-    const rpc::ResourcesData &resources_data,
-    std::unordered_map<google::protobuf::Map<std::string, double>, rpc::ResourceDemand>
-        *aggregate_load) const {
-  auto load = resources_data.resource_load_by_shape();
-  for (const auto &demand : load.resource_demands()) {
-    auto &aggregate_demand = (*aggregate_load)[demand.shape()];
-    aggregate_demand.set_num_ready_requests_queued(
-        aggregate_demand.num_ready_requests_queued() +
-        demand.num_ready_requests_queued());
-    aggregate_demand.set_num_infeasible_requests_queued(
-        aggregate_demand.num_infeasible_requests_queued() +
-        demand.num_infeasible_requests_queued());
-    aggregate_demand.set_backlog_size(aggregate_demand.backlog_size() +
-                                      demand.backlog_size());
-  }
-}
-
 void GcsResourceManager::HandleGetAllResourceUsage(
     rpc::GetAllResourceUsageRequest request,
     rpc::GetAllResourceUsageReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
+  // This is a hack. We use the stale but less inconsistent resource info
+  // from GCS autoscaler state manager instead of our local resource info.
```

Comment:
Where do you use the gcs autoscaler state manager?

Reply:
oops, old comment leftover from earlier revision